File: server/src/main/java/ru/vitaliy/petrov/server/forms/requests/chats/MessageUpdateRequest.java
package ru.vitaliy.petrov.server.forms.requests.chats;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.sql.Date;
import java.sql.Time;

@Data
public class MessageUpdateRequest {

    @NotNull
    private final Date messageUpdateDate;

    @NotNull
    private final Time messageUpdateTime;

    @NotNull
    @NotEmpty
    private final String updatedMessageText;

    @NotNull
    private final Long messageID;
}

File: client/app/src/main/java/com/petrov/vitaliy/caraccidentapp/domain/models/requests/chats/MessageUpdateRequest.kt
package com.petrov.vitaliy.caraccidentapp.domain.models.requests.chats

import java.sql.Date
import java.sql.Time

data class MessageUpdateRequest(
    val messageUpdateDate: Date,
    val messageUpdateTime: Time,
    val updatedMessageText: String,
    val messageID: Long
)
